### PR TITLE
Use in house events for analytics rather than third parties

### PR DESCRIPTION
**Pull request by kujenga**
This PR switches from third party analytics services (KissMetrics and Segment) to our in-house system. Implementing views correctly required reorganizing (and improving the organization of) content within the website. Everything should be rendered just the same, but it makes for a noisy PR.

Query parameter handling might be a bit more specific that is was previously, but it should handle all the types of params we're currently using if my memory isn't failing me. Should be fairly easy to add functionality there if necessary.

### CODE REVIEW

**Comment by jasonbriggs** [comment]
This looks great! How thoroughly did you test it?

File: scripts/events.js [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    }
+
+    return data;
+  };
+
+  return {
+    settings: {
+      // general configuration
+      appName: 'website',
+      baseURI: 'https://web-service-dot-meta-cloud-dev.appspot.com/api',
```
Comment (kujenga) [inline scripts/events.js]: This needs to be customized for production

**Comment by kujenga** [comment]
With the last two additions we should be feature complete against the current KissMetrics setup

**Comment by kujenga** [comment]
@calvinfroedge What are you thoughts on this PR?

**Comment by calvinfroedge** [comment]
@kujenga You asking me to critique your javascript or provide an opinion on whether or not you should be doing custom analytics? If the latter, and you haven't seen [Keen](http://keen.io), I'd check it out. The problem isn't so much tracking / sending events as visualizing them, and Keen is really good at that. 50k events /mo for free...dunno if you're hitting that already or not...it gets kind of pricy after.

If you're asking about the JS, I think you're doing a bit more heavy lifting than you really need to (i.e. using xhr directly ($.post / $.get takes fewer loc), writing your own random id generator (node-uuid, works in browser, and is tiny), and you could be taking advantage of all the wonderful things ES6 has to offer, there are easier ways to work with timestamps, cookie stuff looks kind of hairy...but I mean whatever, if it does the job and it's done.

**Honestly, though, outa context, don't think I can give you really useful advice.**

**Comment by calvinfroedge** [comment]
Just want to add to this...if your goal (probably is), is to make a consistent analytics system across both your product, marketing site, and sysops, then all that probably matters on the client is that you can get the data you need.

It's good to be cognizant of everything you'll need to do to really use that data intelligently once you have it. Maybe not needing to retrieve it from another system will make it easier to use it to change your product behavior, and maybe you don't have to pay a monthly fee, but there are definitely man-hour costs associated with storing, displaying, and utilizing your own analytics system as well. Is it something you want to spend a week on?

**Comment by kujenga** [comment]
Thanks @calvinfroedge! I realize my request was a little vague, but my intent was some quality feedback on the JS code. We looked at (and have been using) a variety of third party tools for this kind of thing, including Keen.io which you mentioned to me a while back. They're all awesome tools, but we currently handle about 3-5 million events per day, at which point they get expensive pretty quickly. We could splinter off the more meaningful events into one of these systems, but that in turn makes it harder to get insights that span all of our data. We're currently piping everything into Google BigQuery, so we'll probably be using one of the [tools](https://cloud.google.com/bigquery/third-party-tools) that integrates with it, e.g. [bime](https://www.bimeanalytics.com/pricing.html) or [tableau](http://www.tableau.com), for a visualization solution down the road. It also makes it easier to have triggers within our system that react to events in real time, as you mentioned. For now though, we already _are_ spending a large fraction of our time on analytics because it's of core importance to our product, so the easier it is to interact with the data in iPython notebooks, the easier it is to do that work. I think those data viz tools can fill in the gaps for us in the future. These larger changes were motivated primarily by the needs of our team. Hopefully that provides a bit of missing context.

There are also benefits from a privacy perspective, as we've already some folks write in to ask about data sharing with third parties, so keeping everything in house makes it easier to have good answers to those questions.

This is actually the first bit of self-contained javascript I've written, besides modifying existing code and little script hacks, so I wanted to get some validation that I'm not making any glaring mistakes. It sounds like there are some rough edges for sure, but good enough to get started with. Is that a fair assessment?

**Comment by kujenga** [comment]
BTW, the id generation is a format we use across all of our services/clients, based on this blog post, so I figured I'd stick to it: https://www.firebase.com/blog/2015-02-11-firebase-unique-identifiers.html

**Comment by jasonbriggs** [comment]
Looks good!

**Pull request merged**